src/Command/Email/ConfigurePlatformEmailCommand.php

```diff
@@ -0,0 +1,331 @@
+    if (count($subscription_applications) === 0) {
+      throw new AcquiaCliException("You do not have access to any applications on the {$subscription->name} subscription");
+    }
+    elseif (count($subscription_applications) === 1) {
+      $applications = $subscription_applications;
+      $this->io->info("You have one application, {$applications[0]->name}, in this subscription.");
+    }
+    else {
+      $applications = $this->promptChooseFromObjectsOrArrays($subscription_applications, 'uuid', 'name', "What are the applications you'd like to associate this domain with? You may enter multiple separated by a comma.", TRUE);
+    }
+
+    $environments_resource = new Environments($client);
+    foreach ($applications as $application) {
+      try {
+        $response = $client->request('post', "/applications/{$application->uuid}/email/domains/{$domain_uuid}/actions/associate");
+        $this->io->success("Domain $base_domain has been associated with Application {$application->name}");
+      } catch (ApiErrorException $e) {
+        // Shows a warning and allows user to continue if the domain has already been associated.
+        // For any other error from the API, the setup will exit.
+        if (strpos($e, 'is already associated with this application') === FALSE) {
+          $this->io->error($e->getMessage());
+          return FALSE;
+        }
+        else {
+          $this->io->warning($e->getMessage());
+        }
+      }
+
+      $application_environments = $environments_resource->getAll($application->uuid);
+      $envs = $this->promptChooseFromObjectsOrArrays($application_environments, 'uuid', 'label', "What are the environments of {$application->name} that you'd like to enable email for? You may enter multiple separated by a comma.", TRUE);
+      foreach ($envs as $env) {
+        try {
+          $response = $client->request('post', "/environments/{$env->uuid}/email/actions/enable");
+          $this->io->success("Platform Email has been enabled for environment {$env->label} for application {$application->name}");
+        }
+        catch (ApiErrorException $e) {
+          // Shows a warning and allows user to continue if Platform Email has already been enabled for the environment.
+          // For any other error from the API, the setup will exit.
+          if (strpos($e, 'is already enabled on this environment') === FALSE) {
+            $this->io->error($e->getMessage());
+            return FALSE;
+          }
+          else {
+            $this->io->warning($env->label . ' - ' . $e->getMessage());
+          }
+        }
+      }
+    }
+    return TRUE;
+  }
+
+  /**
+   * Validates the URL entered as the base domain name.
+   *
+   * @param string $url
+   *
+   * @return string
+   * @throws \Symfony\Component\Validator\Exception\ValidatorException
+   */
+  public static function validateUrl(string $url): string {
+    $violations = Validation::createValidator()->validate($url, [
+      new NotBlank(),
```

Comment:
Anything else to add here? Minimum length? Requires .com or other suffix?

Reply:
Symfony has a URL validator constraint 🤩 adding it in

Reply:
Yeah, but it requires an absolute url, right?

Reply:
Nope - 
<img width="894" alt="Screen Shot 2022-02-20 at 6 33 29 PM" src="https://user-images.githubusercontent.com/28101298/154869308-4074912a-0791-4405-b0dd-225f96c1a8b7.png">

Reply:
ah wait

Reply:
I spoke too soon, re-thinking this

Reply:
Okay, so I ended up setting up a conditional to check whether the URL contains the 'http' scheme, if so it will use the URL validator. Otherwise, Symfony also has a hostname validator that will be used instead. 

I also removed the additional domain input in favor of keeping the validation logic in one place, when the user is prompted.